embedding/embedlite/embedshared/EmbedLiteWindowChild.cpp

```diff
@@ -199,9 +196,42 @@ void EmbedLiteWindowChild::CreateWidget()
               );
   GetWidget()->UpdateBounds(true);
 
+  // Initialize ScreenManager
+  RefreshScreen();
+
   mInitialized = true;
   Unused << SendInitialized();
 }
 
+void EmbedLiteWindowChild::RefreshScreen()
+{
+  LayoutDeviceIntRect rect;
+  if (mRotation == ROTATION_0 || mRotation == ROTATION_180)
+    rect = mBounds;
+  else
+    rect = LayoutDeviceIntRect(0, 0, mBounds.Height(), mBounds.Width());
+
+  AutoTArray<RefPtr<Screen>, 1> screenList;
+  RefPtr<Screen> screen = new Screen(rect, rect, mDepth, mDepth, DesktopToLayoutDeviceScale(mDensity), CSSToLayoutDeviceScale(1.0f), GetWidget()->GetDPI());
+  screenList.AppendElement(screen.forget());
+  ScreenManager::GetSingleton().Refresh(std::move(screenList));
+}
+
+void EmbedLiteWindowChild::SetDepth(int depth)
+{
+  if (depth != mDepth) {
+    mDepth = depth;
+    RefreshScreen();
+  }
+}
+
+void EmbedLiteWindowChild::SetDensity(float density)
```

Comment:
I'd rather have

```
SetScreenProperties(int depth, float density, float dpi)
```
=> That would imply one send from parent to child + only one RefreshScreen()

Reply:
I think this is good enough now. One improvement could be that we'd deliver window properties directly to the window without using view in between. Thanks @mkenttala for adjusting. As said, good enough.